cadence/nfts/blind-edition-nft/BlindEditionNFT.template.cdc

```diff
@@ -53,43 +54,75 @@ pub contract {{ contractName }}: NonFungibleToken {
 
         pub let id: UInt64
 
-        /// The maximum size of this edition.
+        /// The maximum number of NFTs that can be minted in this edition.
         ///
-        pub let size: UInt64
+        /// If nil, the edition has no size limit.
+        ///
+        pub let limit: UInt64?
 
         /// The number of NFTs minted in this edition.
         ///
-        /// The count cannot exceed the edition size.
+        /// This field is incremented each time a new NFT is minted.
+        /// It cannot exceed the limit defined above.
+        ///
+        pub var size: UInt64
+
+        /// The number of NFTs in this edition that have been burned.
+        ///
+        /// This field is incremented each time an NFT is burned.
         ///
-        pub var count: UInt64
+        pub var burned: UInt64
+
+        /// Return the total supply of NFTs in this edition.
+        ///
+        /// The supply is the number of NFTs minted minus the number burned.
+        ///
+        pub fun supply(): UInt64 {
+            return self.size - self.burned
+        }
+
+        /// A flag indicating whether this edition is closed for minting.
+        ///
+        pub var isClosed: Bool
 
         /// The metadata for this edition.
         ///
         pub let metadata: Metadata
 
         init(
             id: UInt64,
-            size: UInt64,
+            limit: UInt64?,
             metadata: Metadata
         ) {
             self.id = id
-            self.size = size
+            self.limit = limit
             self.metadata = metadata
 
-            // An edition starts with a count of zero
-            self.count = 0
+            self.size = 0
+            self.burned = 0
+
+            self.isClosed = false
         }
 
-        /// Increment the NFT count of this edition.
+        /// Increment the size of this edition.
         ///
-        /// The count cannot exceed the edition size.
+        access(contract) fun incrementSize() {
+            self.size = self.size + (1 as UInt64)
+        }
+
+        /// Increment the burn count for this edition.
         ///
-        access(contract) fun incrementCount() {
-            post {
-                self.count <= self.size: "edition has already reached its maximum size"
-            }
+        access(contract) fun incrementBurned() {
```

Comment:
To confirm the assumption here that all these incremental calls will be single-threaded and can't be made from multiple threads, is it correct?

Reply:
Yep that's correct. Cadence doesn't support parallel execution -- it'd be hard to do while still producing a deterministic output, which is required in a blockchain context.